gobblin-core/src/main/java/gobblin/metrics/GobblinMetrics.java

```diff
@@ -238,4 +253,42 @@ private void buildJmxMetricReporter(Properties properties) {
         .convertDurationsTo(TimeUnit.MILLISECONDS).build()));
   }
 
+  private void buildKafkaReporter(Properties properties) {
+    if (!Boolean.valueOf(properties.getProperty(ConfigurationKeys.METRICS_REPORTING_KAFKA_ENABLED_KEY,
+        ConfigurationKeys.DEFAULT_METRICS_REPORTING_KAFKA_ENABLED))) {
+      LOGGER.info("Not reporting metrics to Kafka");
+      return;
+    }
+
+    String brokers = properties.getProperty(ConfigurationKeys.METRICS_KAFKA_BROKERS);
+    String topic = properties.getProperty(ConfigurationKeys.METRICS_KAFKA_TOPIC);
+
+    if (Strings.isNullOrEmpty(brokers)) {
```

Comment:
You can use `Preconditions.checkArgument(properties.containsKey(...))` to check the presence of those two properties.